### DISCUSSION TITLE
Question: How to integrate i18n and [locale] path

### DISCUSSION THREAD

**Discussion by bizzul** [medusajs/nextjs-starter-medusa#270] [Q&A]
Hi There! I need to make a client website with this starter, my question is how to integrate the countryCode middleware and also the locale for setting the language, because here in Switzerland with the same CountryCode (ch) we have 4 languages.

It is possibile?

Thank you!

**Comment by VariableVic** [comment]
Hey! 

It's definitely possible, but it will require some customizations on your end. 

One solution that comes to mind, is replacing the ISO country code by locales like `fr-ch`, `de-ch` and `it-ch`. You can then use the [Accept-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) headers to determine the user's preferred language in middleware.

To link language locales to Medusa regions, you could use the region's `metadata` object, like `region.metadata.locales = ['fr-ch', 'de-ch', 'it-ch']` etc.

Here's some more info on [internationalization in Next.js](https://nextjs.org/docs/pages/building-your-application/routing/internationalization).

Hope this helps, good luck with the project!

**Comment by kb-0912** [comment]
> Hi There! I need to make a client website with this starter, my question is how to integrate the countryCode middleware and also the locale for setting the language, because here in Switzerland with the same CountryCode (ch) we have 4 languages.
> 
> It is possibile?
> 
> Thank you!

Did you find out solution?

**Reply by sseshaheen** [reply]
@kb-0912 @bizzul Have a look at this PR which integrates [next-intl](https://www.npmjs.com/package/next-intl) to the project: https://github.com/medusajs/nextjs-starter-medusa/pull/456. I haven't yet tested your specific request, but it should work fine
